lib/jsonb_accessor/macro.rb

```diff
@@ -65,7 +66,14 @@ def jsonb_accessor(jsonb_attribute, field_types)
           names_and_store_keys.each do |name, store_key|
             define_method("#{name}=") do |value|
               super(value)
-              new_values = (public_send(jsonb_attribute) || {}).merge(store_key => public_send(name))
+
+              attribute_value = public_send(name)
+              # Rails always saves time based on `default_timezone`. Since #as_json considers timezone, manual conversion is needed
+              if attribute_value.acts_like?(:time)
+                attribute_value = (active_record_default_timezone == :utc ? attribute_value.utc : attribute_value.in_time_zone).strftime("%F %R:%S.%L")
```

Comment:
Maybe to be in line with the rest of the code something like `JsonbAccessor::Helpers.active_record_default_timezone` would be better.